packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx

```diff
@@ -0,0 +1,146 @@
+import { NFTInfo } from '@chia-network/api';
+import { usePersistState, useDarkMode } from '@chia-network/core';
+import { t, Trans } from '@lingui/macro';
+import React from 'react';
+import styled from 'styled-components';
+
+import useAllowFilteredShow from '../../hooks/useAllowFilteredShow';
+import useHideObjectionableContent from '../../hooks/useHideObjectionableContent';
+import NFTPreview from '../nfts/NFTPreview';
+import useFilteredNFTs from '../nfts/gallery/NFTfilteredNFTs';
+
+const SearchNFTrow = styled.div`
+  cursor: pointer;
+  width: 100%;
+  padding: 5px;
+  display: table;
+  &:hover {
+    background: ${(props) => (props.isDarkMode ? '#444' : '#eee')} !important;
+  }
+  > div:first-child {
+    width: 50px;
+  }
+  > div {
+    display: table-cell;
+    vertical-align: top;
+    padding: 5px;
+    text-align: left;
+  }
+`;
+
+const NFTSearchedText = styled.div`
+  .highlight {
+    color: ${(props) => props.theme.palette.primary.main};
+  }
+  div:last-child {
+    font-size: 13px;
+  }
+`;
+
+const SearchPlaceholder = styled.div`
+  position: relative;
+  top: 10px;
+  z-index: 3;
+  background: red;
+  > div {
+    background: ${(props) => (props.isDarkMode ? '#222' : '#fafafa')};
+    ::-webkit-scrollbar-thumb {
+      background-color: ${(props) => (props.isDarkMode ? '#444' : '#ddd')};
+      border: 4px solid transparent;
+      border-radius: 8px;
+      background-clip: padding-box;
+    }
+    ::-webkit-scrollbar {
+      width: 16px;
+    }
+    border: 1px solid ${(props) => (props.isDarkMode ? '#222' : '#ddd')};
+    border-radius: 15px;
+    box-shadow: 0px 0px 25px 0 rgba(0, 0, 0, 0.2);
+    max-height: 300px;
+    width: 450px;
+    overflow-x: hidden;
+    overflow-y: hidden;
+    overflow-y: auto;
+    position: absolute;
+    > div:nth-child(odd) {
+      background: ${(props) => (props.isDarkMode ? '#333' : '#fff')};
+    }
+    > div:nth-child(even) {
+      background: ${(props) => (props.isDarkMode ? '#333' : '#fff')};
+    }
+  }
+`;
+
+type OfferBuilderValueSearchProps = {
+  value: string;
+  onSelectNFT: (nftId: string) => void;
+};
+
+function highlightSearchedString(searchString: string, str: string) {
+  if (!str) return '';
+  const r = new RegExp(`(${searchString})`, 'i');
+  return str
+    .split(r)
+    .map((part) =>
+      part.toLocaleLowerCase() === searchString.toLocaleLowerCase() ? (
+        <span className="highlight">{part}</span>
+      ) : (
+        <span>{part}</span>
+      )
+    );
+}
+
+export default function OfferBuilderValueSearch(props: OfferBuilderValueSearchProps) {
+  const { value, onSelectNFT } = props;
+  const [walletId] = usePersistState<number | undefined>(undefined, 'nft-profile-dropdown');
```

Comment:
The last selection in the NFT profile dropdown doesn't seem relevant to the search feature when creating an offer.

Reply:
Tracked as https://github.com/Chia-Network/chia-blockchain-gui/issues/1251